weblate/checks/chars.py

```diff
@@ -39,6 +39,23 @@
 FRENCH_PUNCTUATION = {";", ":", "?", "!"}
 
 
+class AcceleratorKeyCheck(TargetCheck):
+    """Check for inconsistent accelerator keys."""
+
+    check_id = "accelerators"
+    name = _("Accelerator key")
+    description = _("Source and translation do not both contain an accelerator key")
+    default_disabled = True
+
+    def _check_accelerator(self, source, target, key):
+        return (target.count(key) > 1) or (source.count(key) != target.count(key))
```

Comment:
Gettext check behaves slighly differently:

* We are only interested in msgids that contain exactly one '&'. 
* Count the number of '&' in msgstr, but ignore '&&'. 

See https://git.savannah.gnu.org/gitweb/?p=gettext.git;a=blob;f=gettext-tools/src/msgl-check.c;h=32c21adf546cae496b0dcccf6cc235773b8a30e8;hb=HEAD#l708

Reply:
I was trying to catch the case that more than one accelerator is in a translation, but this may trigger a false positive on some markup like _underscores around this text_.

Reply:
Is there really something which uses two accelerators in the string? The Gettext implementation clearly looks only for one.

Reply:
Never, this is why I wanted to flag it as a possible error. If the source or translation has more than one accelerator, there's probably something wrong. This check will pass if there's exactly zero or one accelerator in source and translation, ignoring escapes, because we can't know them all.

Reply:
The gettext logic is more complex:

- it doesn't check strings where source doesn't have single accelerator
- it supports escaping (I guess, it ignores double accelerator)

Reply:
The translationhouse docs that I linked below, mention more than one way of escaping the accelerator. All these depend on the client renderer and are independent of the translation file format. If there are still other ways of escaping, we don't know them. This is why I suggested to ignore escaping and silence false positives using the flag.
OTOH if we pass the accelerator as parameter, as discussed below, we could also pass the escaped accelerator.
The default could be `&` and `&&`, which probably covers most of the cases.
But I could be wrong, because https://github.com/WeblateOrg/weblate/issues/6142 and https://github.com/Gnucash/gnucash/pull/1039 both explicitely mention the `_` as accelerator.

Reply:
I've no real-world experience with this - I'm just describing what Gettext does here, as the original issue was to mimic its behavior. So, it doesn't have to work the way I describe, it just should be considered as somebody took the effort to implement it this way there.

Reply:
Yes, to mimic the gettext behaviour was the original intent. Since then we learned that there's more to it and gettext is just one approach, so I'd like to implement an "as good as possible" solution without overcomplicating the implementation/configuration.